api/v1/storagecluster_types.go

```diff
@@ -97,8 +97,8 @@ type StorageClusterSpec struct {
 	AllowRemoteStorageConsumers bool `json:"allowRemoteStorageConsumers,omitempty"`
 
 	// ProviderAPIServerServiceType Indicates the ServiceType for OCS Provider API Server Service.
-	// The supported values are NodePort or LoadBalancer. The default ServiceType is NodePort if the value is empty.
-	ProviderAPIServerServiceType corev1.ServiceType `json:"providerAPIServerServiceType,omitempty"`
+	// The default ServiceType is derived from hostNetwork field.
+	ProviderAPIServerServiceType *corev1.ServiceType `json:"providerAPIServerServiceType,omitempty"`
```

Comment:
Why do we need a pointer when service type can be set to an empty string?

Reply:
changed to empty string, used a pointer as it was mentioned in one of the meetings.